src/server/pfs/s3/s3.go

```diff
@@ -64,6 +64,10 @@ func (c *controller) requestClient(r *http.Request) *client.APIClient {
 	vars := mux.Vars(r)
 	if vars["s3gAuth"] != "disabled" {
 		accessKey := vars["authAccessKey"]
+		if strings.HasPrefix(accessKey, "PAC1") {
```

Comment:
To be honest, I'm offhand skeptical of this logic from a product perspective, so I want to make sure I understand and see what you think of a possible alternative approach.

- My skepticism is rooted in the fact that the auth token has been a point of confusion for users in the past (e.g. with auth off, the ACCESS_KEY_ID and SECRET_ACCESS_KEY _must_ be nonempty and _must_ match, but otherwise can be any string, which users find confusing). If we additionally have to explain that it _must_ contain this fairly opaque prefix, I worry that it'll cause even more confusion
- An alternative approach could be to put the project in the bucket name. Currently, in the global S3 gateway, buckets are identified as `<branch>.<repo>`, e.g. `master.myrepo`. What could work is to implement the convention that if S3G bucket names contain a single dot, then they refer to branches in the default project. If they contain two dots, then the identity becomes `<branch>.<repo>.<project>`. That way, existing keys and bucket names continue to work as they did previously, and things happen in the default project by default, but support for other projects is present.
- For the sidecar S3 gateway, buckets exist in the context of a pipeline, and I believe buckets are identified with the name of the PFS input they correspond to. So `/pfs/foo` becomes `s3://foo`. I believe this convention is already robust in the presence of projects (where inputs from other projects still have an input name, which is the repo name by default) so nothing needs to change.

WDYT? Do you prefer the auth access-key based approach or are you ok with the alternative? And let us know if the issue with the alternative approach is bandwidth

Reply:
I see. I think what may make sense is to merge this PR to the end of unblocking projects, but for Integrations to plan on reverting the `PAC1` access key behavior and implementing the bucket name approach described blow (with a solution to the ambiguity you raise) before the 2.5 release. If we're quick, we'll have the solution proposed below, but if we run out of time, we'll still be able to release 2.5 with project support.

> Unfortunately, the currently accepted bucket name pattern is actually [<commitID>.][<branch>. | <branch>.<type>.]<repoName>

I think we should change the scheme to replace `<repo>.<type>` with `<repo>_<type>` to make the three-dots approach work (IIRC, types apply to repos rather than branches, e.g. spec and meta. If they do apply to branches, the fix still works, but will be slightly different). I know it's an API-incompatible change, but a) I'd bet money that zero people have ever accessed a non-user repo via the S3 gateway, so b) foisting a breaking change on those people seems basically way better than foisting a magic constant on the (I expect) significant number of users interested in projects.

> Unfortunately, our s2 library doesn’t just use SecretKey to deal with access key IDs; it also passes it in the authAccessKey request variable. This bug was revealed when testing a project-aware S3 sidecar client.

This I would like to understand--why is it a bug for S2 to pass the secret key in authAccessKey? (separately, we'd like to bring S2 into the main pach repo and archive the S2 repo, among other S2 changes).

I'm approving the PR, but please also get approval from Ben or Sahith before merging.